From 0da347aae6e66e95c7dfa847598798c6de1db292 Mon Sep 17 00:00:00 2001
From: Max Maischein <corion@corion.net>
Date: Tue, 22 Mar 2011 21:56:46 +0100
Subject: [PATCH 2/5] Add a sanity self-check for have_functions()

---
 Makefile.PL |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/Makefile.PL b/Makefile.PL
index 26ab13e..b966faa 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -104,6 +104,9 @@ REASON
 # missing functions with croaking stubs. 
 # We also store the list of available functions in a file for skipping the 
 # corresponding tests. 
+my $sanity_check = have_functions('pcap_open_live');
+die "Couldn't find pcap_open_live(). Something is very wrong."
+    unless $sanity_check;
 my @funcs = have_functions(find_functions());
 $options{DEFINE} .= cpp_defines(@funcs);
 open(FUNCS, '>funcs.txt') or warn "warning: can't write 'funcs.txt': $!\n";
-- 
1.6.5.1.1367.gcd48

